vllm/entrypoints/openai/api_server.py

```diff
@@ -560,6 +560,18 @@ def init_app_state(
     )
 
 
+def create_server_socket(addr: Tuple[str, int]) -> socket.socket:
+    family = socket.AF_INET
+    if ":" in addr[0]:
```

Comment:
please use the builtin library `ipaddress` to tell if an address is v6.

Reply:
Its even better to use socket.getaddrinfo:
```python    
host = args.host or ""
address_family = socket.getaddrinfo(host, args.port)[0][0]
sock = socket.socket(address_family, socket.SOCK_STREAM)
sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
sock.bind((host, args.port))
```

Reply:
for ipv6 address detect I followed uvicorn's impl (see https://github.com/encode/uvicorn/pull/837), btw I tested `socket.getaddrinfo` and it fails on my linux when `host == ""`, but works on Windows and macOS

Reply:
What you have (and in uvicorn) is valid, too, though it can accidentally misinterpret an invalid IP string, like `127.0.0.1:8088`.

You can use `vllm.utils.is_valid_ipv6_address` as a shortcut to implement @youkaichao 's suggestion.

Reply:
fixed by using `vllm.utils.is_valid_ipv6_address` and thanks :)